haystack/modeling/model/adaptive_model.py

```diff
@@ -637,7 +637,9 @@ def load(cls, load_dir: Union[str, Path], device: str, **kwargs):  # type: ignor
         sess_options.graph_optimization_level = onnxruntime.GraphOptimizationLevel.ORT_ENABLE_EXTENDED
         # Use OpenMP optimizations. Only useful for CPU, has little impact for GPUs.
         sess_options.intra_op_num_threads = multiprocessing.cpu_count()
-        onnx_session = onnxruntime.InferenceSession(str(load_dir / "model.onnx"), sess_options)
+
+        providers = kwargs.get("providers", ["CPUExecutionProvider"] if device.type == "cpu" else ["CUDAExecutionProvider"])
```

Comment:
Just a small thing that also came up in the tests run by the CI:
`haystack/modeling/model/adaptive_model.py:641: error: "str" has no attribute "type"`
The type annotations of the `load` method suggest that `device` is of type `str` and therefore has no `.type` attribute. Could you maybe test with comparing directly to the string, like `device == "cpu"`? Otherwise looks good to me. 
An alternative to `else ["CUDAExecutionProvider"])` would have been to list multiple providers sorted by priority, e.g., `['CUDAExecutionProvider', 'CPUExecutionProvider']` but I think it's better to make it explicit here and let it fail otherwise.

Reply:
The type annotation on the method is either incorrect or this is an existing problem with other callers of this method...  

What ends up passed in is a `torch.device`.  I don't know if this intended, and/or a bug, but `Inferencer.load`  passes in  a `torch.device` when calling this method.  (which makes `AdaptiveModel.load` type annotations  similarity incorrect).  

So there may be some larger issues here that need to be cleaned up/fixed beyond the scope of this small PR.   Let me know how you want to proceed here to resolve the CI error....  I could change the type annotation to be a `torch.device` just in `ONNXAdapativeModel`.  Or if that is not desired,  would you want to suppress/ignore the type check on this line?